### DISCUSSION TITLE
Should I make repositories for every code I write

### DISCUSSION THREAD

**Discussion by adhyan-jain** [community/community#143205] [New to GitHub] [answered]
### Select Topic Area

Question

### Body

I am a beginner in programming and wanted to know if I should make repositories for all my programs that I do for practice. The point is , is there any merit in doing that? Some say it's your choice and it doesn't matter much and some say that it shows your consistency and it is encouraged.

**Comment by davevad93** [comment]
Hi @adhyan-jain ,

It depends, for example if your projects are independent, keep them in separate repos, don't put them all in a single repo. In the end the choice is yours, but personally I suggest you to push your code to GitHub whether it's something simple or not, thus you also gain experience with version control.

**Comment by joegeorge022** [comment]
Creating repositories for each piece of code can be helpful if the project is something you plan to revisit, share, or build upon later. Here’s a breakdown of when it’s a good idea:

## 1. **Projects or Tutorials**: For any full project, experiment, or tutorial that’s complete or complex, a dedicated repository helps keep code organized and easily shareable.

## 2. **Reusability**: If you think you’ll reuse the code in other projects, hosting it on GitHub or another platform makes it easily accessible.

## 3. **Collaborative Projects**: For group projects or contributions, it’s essential to have a repository to manage changes and allow for collaboration.

## 4. **Portfolio Building**: Repositories are a great way to showcase skills and progress, so having more repositories can demonstrate your experience in different areas.

For smaller snippets or code you don’t intend to keep long-term, you might store them locally or in a single "sandbox" or "practice" repository. That way, you avoid cluttering GitHub while still keeping useful code accessible.

**Comment by ravindudil5han** [comment, marked as answer]
🌱 Hey there! As a beginner, you’re on the right track by thinking about how best to document your programming journey! Creating repositories for each practice program has its advantages, especially in terms of growth and visibility. Here’s why it can be beneficial:

📈 Shows Progress & Consistency: By regularly uploading your practice code, you’re showcasing your learning journey and commitment. It’s a great way to let others (including potential collaborators or employers) see your steady progress over time.

📚 Organized Learning Record: Having a repository of your programs can help you keep track of what you've learned. Later, if you want to revisit concepts, all your code is neatly organized and easily accessible.

🚀 Get Feedback & Contributions: Other developers or mentors can provide feedback on your code, which is invaluable for learning. Open repositories also make it easier for you to engage with the GitHub community and maybe even gain some stars! ⭐

🛠️ Builds Practical Git Skills: Frequent use of Git and GitHub, even for small projects, builds your practical experience with version control, which is essential for software development.

🔑 In Summary: While it’s optional, creating repositories for each practice project is encouraged! It’ll help you build habits that support growth in your skills and online presence. But ultimately, the choice is yours! 😊

Good luck and happy coding! 💻✨

**Comment by ghostinhershell** [comment]
Hi @adhyan-jain 👋🏾, if any of the responses here answered your question can you please mark their response as the answer? When you mark a question as an answer, GitHub will highlight the comment and replies to the comment to help visitors quickly find the answer.
![image](https://github.com/user-attachments/assets/b89e7161-c2d1-47e2-a058-242df633e6b2)

Thanks!

**Comment by siba-9439** [comment]
Hi @adhyan-jain 

First of all segregate the repos according to the project, then if you are thinking that your code is really necessary/helpful for your project then push it to github or else save that locally. If you have multiple variation or dependency on a same project then put that all thing into one repo by creating the new files.

I think this is helpful to you. For more thing you can connect with me in LinkedIn by visiting my profile.

**Comment by Kitty-777** [comment]
You don’t necessarily need to create a new repository for every piece of code you write. It depends on the purpose and organization of your work. If each piece of code represents a unique project-for example: new-app,library or tool,create a separate repository. This makes it easy to manage.

**Comment by NyxianNetwork** [comment]
Creating repositories for your practice programs can be beneficial, especially as a beginner. Here’s why:

1. **Track Progress**: You’ll be able to look back and see how much you’ve improved over time, which is encouraging and valuable.
2. **Build a Portfolio**: Even small projects demonstrate your learning process, consistency, and growing skills, which can be useful if you want to showcase your progress later on.
3. **Version Control Skills**: Working with Git and GitHub regularly helps you get comfortable with version control, which is essential in real-world programming.

However, if you're not comfortable sharing everything, you can make some repositories private or only upload more polished projects. The key is consistency and using GitHub to support your learning.
